src/python/pants/util/dirutil.py

```diff
@@ -339,8 +348,9 @@ def safe_concurrent_rename(src, dst):
   if os.path.isdir(src):  # Note that dst may not exist, so we test for the type of src.
     safe_rmtree(dst)
   else:
-    safe_delete(dst)
+    safe_rmtree(dst, recursive=True) if os.path.islink(src) and recursive else safe_delete(dst)
```

Comment:
I would invert this, so that the more common case appears first.

```python
safe_delete(dst) if not os.path.islink(src) or not recursive else safe_rmtree(dst, recursive=True)
```